spike/CodeReaderDemo.py

```diff
@@ -1,100 +1,254 @@
-port = ''
-operation = ''
-number = ''
-speed = 0
-angle = 0
-
-def main():
-
-    global port
-    global operation
-    global number
-    
-    #Open racer file
-    racer = open("Racer.txt", "r")
-
-    #Define symbols
+class CodeParser():
+
+    #Parsing variables
+    PORT = ''
+    OPERATION = ''
+    NUMBER = ''
+    JUMP = ''
+    FUNC = ''
+    T1 = ''
+    T2 = ''
+    #FUNCTIONS = []
+
+    path = ''
+    SPEED = 0
+    ANGLE = 0
+
+    lineNum = 0
+
+    #Parsing libraries (or "keywords" if you wanna not sound like an arrogant prick)
     OPERATIONS = ['+', '-', '*', '/', '=']
     NUMBERS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
     WHITESPACE = ' '
     PORTS = ["THROTTLE", "TURN"]
-    
-    #Break down symbols together
-    lexeme = ''
-    if racer.mode == "r":
-        code = racer.readlines()
-        for line in code:
-            port = ''
-            operation = ''
-            number = ''
-            lexeme = ''
-            for i, char in enumerate(line):
-                if char != WHITESPACE:
-                   lexeme += char
-                if (i + 1 < len(line)):
-                    if line[i + 1] == WHITESPACE or line[i + 1] == '\n':
-                        if lexeme in PORTS:
-                            port = lexeme
-                        if lexeme in OPERATIONS:
-                            operation = lexeme
-                        if bool([ele for ele in NUMBERS if(ele in lexeme)]):
-                            number = lexeme
-                        lexeme = ''
-                if (i == len(line) - 1):
-                    if lexeme in PORTS:
-                        port = lexeme
-                    if lexeme in OPERATIONS:
-                        operation = lexeme
-                    if bool([ele for ele in NUMBERS if(ele in lexeme)]):
-                        number = lexeme
-                    lexeme = ''
+
+
+    #Initialize the CodeParser
+    def __init__(self, path):
+        self.path = "Racer.txt"
+        self.SPEED = 0
+        self.ANGLE = 0
+
+        #Open racer file
+        self.racer = open(self.path, "r")
```

Comment:
This doesn't have to be a `self` variable since we only use it temporarily to read in the file